lib/promscrape/scraper.go

```diff
@@ -390,25 +401,25 @@ func equalLabel(a, b *prompbmarshal.Label) bool {
 	return true
 }
 
-// runScrapeWorkers runs sws.
-//
-// This function returns after closing stopCh.
-func runScrapeWorkers(sws []ScrapeWork, pushData func(wr *prompbmarshal.WriteRequest), stopCh <-chan struct{}) {
-	tsmGlobal.RegisterAll(sws)
-	var wg sync.WaitGroup
-	for i := range sws {
-		cfg := &sws[i]
-		c := newClient(cfg)
-		var sw scrapeWork
-		sw.Config = *cfg
-		sw.ReadData = c.ReadData
-		sw.PushData = pushData
-		wg.Add(1)
-		go func() {
-			defer wg.Done()
-			sw.run(stopCh)
-		}()
+func hashScrapeWork(sw ScrapeWork) (string, error) {
```

Comment:
bug here, as bson encode can not get un export value, some change such as cert will not be watched